### ISSUE TITLE
Improve Process rejection error message

### ISSUE THREAD

**Issue by Zelldon** [camunda-cloud/zeebe#6740]
**Description**

If a user deploys a process model with unsupported elements it gets a quite cryptic failure message. This is then also printed in the brokers log.

Example output
```
 
Expand all | Collapse all{
 insertId: "1vpgyu7g4jgwbzq"  
 jsonPayload: {
  @type: "type.googleapis.com/google.devtools.clouderrorreporting.v1beta1.ReportedErrorEvent"   
  context: {…}   
  exception: "java.lang.ClassCastException: class org.camunda.bpm.model.xml.impl.instance.ModelElementInstanceImpl cannot be cast to class io.zeebe.model.bpmn.instance.BpmnModelElementInstance (org.camunda.bpm.model.xml.impl.instance.ModelElementInstanceImpl and io.zeebe.model.bpmn.instance.BpmnModelElementInstance are in unnamed module of loader 'app')
	at java.util.ArrayList.forEach(Unknown Source) ~[?:?]
	at io.zeebe.model.bpmn.traversal.ModelWalker.walk(ModelWalker.java:62) ~[zeebe-bpmn-model-1.0.0-alpha4.jar:1.0.0-alpha4]
	at io.zeebe.engine.processing.deployment.transform.BpmnValidator.validate(BpmnValidator.java:40) ~[zeebe-workflow-engine-1.0.0-alpha4.jar:1.0.0-alpha4]
	at io.zeebe.engine.processing.deployment.transform.DeploymentTransformer.transformResource(DeploymentTransformer.java:113) ~[zeebe-workflow-engine-1.0.0-alpha4.jar:1.0.0-alpha4]
	at io.zeebe.engine.processing.deployment.transform.DeploymentTransformer.transform(DeploymentTransformer.java:90) ~[zeebe-workflow-engine-1.0.0-alpha4.jar:1.0.0-alpha4]
	at io.zeebe.engine.processing.deployment.DeploymentCreateProcessor.processRecord(DeploymentCreateProcessor.java:89) ~[zeebe-workflow-engine-1.0.0-alpha4.jar:1.0.0-alpha4]
	at io.zeebe.engine.processing.streamprocessor.TypedRecordProcessor.processRecord(TypedRecordProcessor.java:54) ~[zeebe-workflow-engine-1.0.0-alpha4.jar:1.0.0-alpha4]
	at io.zeebe.engine.processing.streamprocessor.ProcessingStateMachine.lambda$processInTransaction$3(ProcessingStateMachine.java:296) ~[zeebe-workflow-engine-1.0.0-alpha4.jar:1.0.0-alpha4]
	at io.zeebe.db.impl.rocksdb.transaction.ZeebeTransaction.run(ZeebeTransaction.java:84) ~[zeebe-db-1.0.0-alpha4.jar:1.0.0-alpha4]
	at io.zeebe.engine.processing.streamprocessor.ProcessingStateMachine.processInTransaction(ProcessingStateMachine.java:286) ~[zeebe-workflow-engine-1.0.0-alpha4.jar:1.0.0-alpha4]
	at io.zeebe.engine.processing.streamprocessor.ProcessingStateMachine.processEvent(ProcessingStateMachine.java:254) ~[zeebe-workflow-engine-1.0.0-alpha4.jar:1.0.0-alpha4]
	at io.zeebe.engine.processing.streamprocessor.ProcessingStateMachine.tryToReadNextEvent(ProcessingStateMachine.java:219) ~[zeebe-workflow-engine-1.0.0-alpha4.jar:1.0.0-alpha4]
	at io.zeebe.engine.processing.streamprocessor.ProcessingStateMachine.readNextEvent(ProcessingStateMachine.java:210) ~[zeebe-workflow-engine-1.0.0-alpha4.jar:1.0.0-alpha4]
	at io.zeebe.util.sched.ActorJob.invoke(ActorJob.java:76) [zeebe-util-1.0.0-alpha4.jar:1.0.0-alpha4]
	at io.zeebe.util.sched.ActorJob.execute(ActorJob.java:39) [zeebe-util-1.0.0-alpha4.jar:1.0.0-alpha4]
	at io.zeebe.util.sched.ActorTask.execute(ActorTask.java:122) [zeebe-util-1.0.0-alpha4.jar:1.0.0-alpha4]
	at io.zeebe.util.sched.ActorThread.executeCurrentTask(ActorThread.java:94) [zeebe-util-1.0.0-alpha4.jar:1.0.0-alpha4]
	at io.zeebe.util.sched.ActorThread.doWork(ActorThread.java:78) [zeebe-util-1.0.0-alpha4.jar:1.0.0-alpha4]
	at io.zeebe.util.sched.ActorThread.run(ActorThread.java:191) [zeebe-util-1.0.0-alpha4.jar:1.0.0-alpha4]
"   
  logger: "io.zeebe.broker.process"   
  message: "Unexpected error while processing resource 'blocking.bpmn'"   
  serviceContext: {…}   
  thread: "Broker-1-zb-actors-0"   
 }
```

For a model which contains user task.

![blocking](https://user-images.githubusercontent.com/2758593/113831643-1d6d9780-9788-11eb-81f8-f8b8314ab1fb.png)
[blocking.bpmn.txt](https://github.com/camunda-cloud/zeebe/files/6269967/blocking.bpmn.txt)

**Comment by Zelldon** [comment]
Was reported by our error reporting tool https://console.cloud.google.com/errors/CKze6q-Xpo_S8QE?service=zeebe&time=P7D&project=camunda-cloud-240911&authuser=1

**Comment by saig0** [comment]
This issue has the same root cause as #4817.

**Comment by npepinpe** [comment]
Closed as duplicate #4817
